### ISSUE TITLE
TimescaleDB functions are not recognized ('does not exist')

### ISSUE THREAD

**Issue by Wafje** [supabase/supabase#20317]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

After a restart of my supabase project, the timescaledb extention is not working properly anymore.
When executing a query to create a table, the DB does not recognize the timescaledb functions anymore.

When setting up a new supabase project, this is also not working anymore.

## To Reproduce

1. Create new project,
2. Enable TimescaleDB extension via webUI
3. Run query via webUI

```
CREATE TABLE conditions (
   time        TIMESTAMPTZ       NOT NULL,
   location    TEXT              NOT NULL,
   device      TEXT              NOT NULL,
   temperature DOUBLE PRECISION  NULL,
   humidity    DOUBLE PRECISION  NULL
);

SELECT create_hypertable('conditions', by_range('time'));
```
4. `ERROR:  42883: function by_range(unknown) does not exist` is returned.

When dropping the by_range function from the query. A similar error on create_hypertable is returned.

## Expected behavior

Return of created hypertable

## System information

- OS: Ubuntu 20.04, Windows
- Browser (if applies) Microsoft Edge, Chrome

**Comment by KevinBrolly** [comment]
> After a restart of my supabase project, the timescaledb extention is not working properly anymore.

Hey @Wafje Did you experience this on an older project that you had unpaused or on a brand new project on the supabase platform?

**Comment by Wafje** [comment]
Hi @KevinBrolly,
On both actually. 

Initially on an older project (3ish months old) where I was experimenting with postgres _realtime_ in combination with Timescale tables, that caused a crash on the project (I think), that restarted it automatically. From that point I dropped my tables and ran my schema creation queries and to start fresh. There I noticed the issue.

I have also created a blank project and the results were the same. Meaning the issue is easily reproduceable and project independent.

**Comment by Wafje** [comment]
> Edit: I will connect with pgAdmin / psql to the DB and to see if I have the same results.

Same issue.

**Comment by GaryAustin1** [comment]
I installed the timescaledb extension and just ran the code from here https://supabase.com/docs/guides/database/extensions/timescaledb?database-method=dashboard
This has the same function and works.
![image](https://github.com/supabase/supabase/assets/54564956/58aadd4d-0d3b-4042-819e-1932825d04bc)
Tested on Postgres version 15.1.0.155 and the latest 15.1.1.2 
Did you install the extension to the extensions schema?

**Comment by GaryAustin1** [comment]
I notice you use TIMESTAMPTZ  instead of the demo's TIMESTAMP.  If the functions are expecting TIMESTAMP they would say not found if you used TIMESTAMPTZ.  I don't know if they are or not as I don't use timescaledb.

**Comment by Wafje** [comment]
Hi @GaryAustin1,
Thanks for testing on your end.

I was following this example, which includes the by_range function.
https://docs.timescale.com/use-timescale/latest/hypertables/create/

Looking at the postgres docs, it drops the by_range function.
https://supabase.com/docs/guides/database/extensions/timescaledb

This way it also works on my end.
Timescale mentions this in their docs: 'The by_range dimension builder is an addition to TimescaleDB 2.13.'
While supabase supports version 2.9.1.

This is a mistake on my end, something must have been wrong with my original schema.
Thanks for the support!

**Comment by GaryAustin1** [comment]
Ah.  I did not go down that path as I thought you said it got the same error without it.
Glad it is resolved.

**Comment by thatkit** [comment]
I'm not sure my comment is of any relevance to your issue, buy I've noticed the same error running [TimescaleDB from docker](https://docs.timescale.com/self-hosted/latest/install/installation-docker/). My mistake was that I created an additional database "db" to a pre-existing one "postgres".

So, the thing is TimescaleDB integrates its extensions on the default "postgres" database, but not to those you've created. I could check this by running `\dx` inside of each db. And this is obviously the TimescaleDB extension that allows us to use `by_range`.

Everything worked just fine, when I switched to that pre-generated-for-me "postgres" database.
